src/exo/stdlib/scheduling.py

```diff
@@ -269,19 +264,26 @@ def _replace_helper(proc, subprocs, mem_aware, once):
         i = 0
         while True:
             try:
+                block = proc.find(f"{pattern} #{i}").expand(0, len(body) - 1)
+                assert len(block) == len(
+                    body
+                ), "block is too close to the end of the body"
```

Comment:
@skeqiqevian 
Is this supposed to be an error when the length of a block from the pattern is less than the length of the subproc body? If so, shouldn't this be SchedulingError? Try/catching the `AssertionError` in the code below may hide various internal compiler errors

Reply:
As it currently works, `expand` will automatically clamp to the end of the scope. So if `proc.find` returns a statement at the end of a body, and we try to expand it by 2, we will receive a block of size 1. Maybe this should be a type of UnificationError?

Reply:
Yeah I think `UnificationError` is reasonable here. If you change that, can you remove catching `AssertionError` from below?